python/graphscope/gsctl/commands/common.py

```diff
@@ -82,5 +82,70 @@ def close():
             info("Try 'gsctl --help' for help.")
 
 
+disclaimer = """
+Disclaimer: The `estimate` command is a roughly estimation of the memory usage of the graphscope engine.
+The actual memory usage may vary vastly due to the complexity of the graph algorithm and the data distribution.
+
+The estimation is based on the following formulas:
+let #V = number of vertices, #E = number of edges,
+assume load_factor of hashmap is 0.41, sizeof(inner_id) = sizeof(edge_id) = sizeof(vertex_id) = 8 bytes,
+where vertex_id is the original representation of vertex, it could be string though, in this case, the memory usage will grow.
+Graph:
+    1. vertex
+        - vertex map: #V * (sizeof(vertex_id) + sizeof(inner_id)) * (1 / load_factor)
+        - vertex table: size of vertex files in uncompressed CSV format, unit is GB
+    2. edge
+        - CSR + CSC: 2 * #E * (sizeof(inner_id) + sizeof(edge_id))
+        - offset array: #V * sizeof(size_t)
+        - edge table: size of vertex files in uncompressed CSV format, unit is GB
+    3. additional data structure when graph is partitioned:
+        - outer vertex ID to local ID map:
+        - local ID to outer vertex ID array:
+
+This is the minimum memory usage of the graphscope engine with 1 partition, the actual memory usage would be larger than this estimation.
+User should enlarge it by a multiple it by some factor (e.g. 2) to simulate the memory usage of the graphscope engine.
+"""
+
+
+def estimate_gae_memory_usage(v_num, e_num, v_file_size, e_file_size):
+    gb = 1024 * 1024 * 1024
+    # vertex map
+    vertex_map = v_num * (8 + 8) * (1 / 0.41) / gb
+    # vertex table
+    vertex_table = v_file_size
+    # edge
+    edge = 2 * e_num * (8 + 8) / gb
+    # offset array
+    offset_array = v_num * 8 / gb
+    # edge table
+    edge_table = e_file_size
+    return vertex_map + vertex_table + edge + offset_array + edge_table
+
+
+@cli.command()
+@click.option(
+    "-s",
+    "--storage",
```

Comment:
-e and --engine?

Reply:
e is used for enum. Seems the usage is more relate to storage.